### PR TITLE
Feature/add statsclient

### PR DESCRIPTION
**Pull request by rwhitten577**
Brings over StatsClient classes and the Lambda types from nights-watch repo. 

Right now this gets imported as: 
```
from omelette.statsd import AbstractStatsClient, get_stats_client
```

I tried first with `omelette.monitoring.statsd` but without anything else planned to go in that namespace, it seemed a bit verbose. If we are keeping our ABCs and their concrete children in one file, I'm wondering how much we need to namespace. E.g. to take one of your examples, we could have: 
```
from omelette.storage import AbstractStorage, S3Storage, SFTPStorage, LocalStorage
``` 
where it's still namespaced by file name and clearly marked as "storage". Just a thought — I could be convinced otherwise if you feel strongly about it.

### CODE REVIEW

**Comment by JoshuaSchlichting** [comment]
I would anticipate that if "storage" objects became more plentiful in the future, that they would end up being broken down into smaller modules that are a part of a `storage` namespace that isn't necessarily a `.py` module. (Though, this change would not be "felt" by downstream clients of the library.)

With this in mind, and the point your making about minimal packaging up front: what if we renamed your `statsd.py` module to `monitoring.py`, effectively making that namespace `monitoring`. Then, in the future, we can add non "statsd" monitoring to the namespace, and then grow that out into more modules in a `monitoring` package if the library ever grows in that direction? We open up the chance to add code to this namespace without changing namespaces.

Thoughts?

**Comment by rwhitten577** [comment]
> ...what if we renamed your `statsd.py` module to `monitoring.py`, effectively making that namespace `monitoring`. Then, in the future, we can add non "statsd" monitoring to the namespace, and then grow that out into more modules in a `monitoring` package if the library ever grows in that direction? We open up the chance to add code to this namespace without changing namespaces.
> 

Sounds good to me! Just pushed that update

File: omelette/settings.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 from dotenv import find_dotenv, load_dotenv
 from tomlkit import parse, items
 
-from omelette.core import PROJECT_ENV
+from omelette.constants import PROJECT_ENV
 
 logger = logging.getLogger(__name__)
```
Comment (JoshuaSchlichting) [inline omelette/settings.py:12]: One last thing before I "click the button":

Can we make this `logger` -> `_logger`?

**Review by JoshuaSchlichting — APPROVED** [review]
LGTM

**Pull request merged**